examples/simon.py

```diff
@@ -0,0 +1,339 @@
+'''
+---------------------------------
+SIMONS'S ALGORITHM - OVERVIEW
+---------------------------------
+Simon's Algorithm solves the problem of finding a particular value of s when some function f:{0, 1}^n --> {0, 1}^n
+is inputted into the program that follows this rule: "f(x) = f(y) if and only if x (+) y is in the set {0^n, s}"
+---------------------------------
```

Comment:
Delete this `STEPS` section and replace it with a reference to Simon's Paper:  https://epubs.siam.org/doi/10.1137/S0097539796298637